common/src/main/java/com/microsoft/identity/common/internal/result/MsalBrokerResultAdapter.java

```diff
@@ -392,38 +393,40 @@ private ServiceException getServiceException(@NonNull final BrokerResult brokerR
     }
 
     /**
-     * Get the bundle from the AccountRecord list.
-     * @param records List of AccountRecord
+     * Get the bundle from the ICacheRecord list.
+     * @param records List of ICacheRecord
      * @return Bundle
      */
-    public Bundle bundleFromAccountRecordList(@NonNull final List<AccountRecord> records) {
+    public Bundle bundleFromCacheRecordList(@NonNull final List<ICacheRecord> records) {
         final Bundle resultBundle = new Bundle();
-        ArrayList<String> accountRecordString = new ArrayList<>();
-        for (AccountRecord record : records) {
-            final String recordInGson = new Gson().toJson(record, AccountRecord.class);
-            accountRecordString.add(recordInGson);
+        ArrayList<String> cacheRecordString = new ArrayList<>();
+        if (records != null && !records.isEmpty()) {
+            for (ICacheRecord record : records) {
+                final String recordInGson = new Gson().toJson(record, CacheRecord.class);
+                cacheRecordString.add(recordInGson);
+            }
         }
 
-        resultBundle.putStringArrayList(BROKER_ACCOUNTS, accountRecordString);
+        resultBundle.putStringArrayList(BROKER_ACCOUNTS, cacheRecordString);
         return resultBundle;
     }
 
     /**
-     * Get the AccountRecord list from bundle.
+     * Get the CacheRecord list from bundle.
      * @param bundle Bundle
-     * @return List of AccountRecord
+     * @return List of CacheRecord
      */
-    public static List<AccountRecord> getAccountRecordListFromBundle(@NonNull final Bundle bundle) {
+    public static List<ICacheRecord> getCacheRecordListFromBundle(@NonNull final Bundle bundle) {
         final ArrayList<String> accountsList = bundle.getStringArrayList(BROKER_ACCOUNTS);
-        final List<AccountRecord> result = new ArrayList<>();
-        if (accountsList == null) {
+        final List<ICacheRecord> result = new ArrayList<>();
+        if (accountsList == null || accountsList.isEmpty()) {
```

Comment:
`null` here seems distinctly different than empty....

Receiving `null` in this case would signal a bug, correct? If yes, let's `throw` some meaningful Exception.

Empty to me says: no accounts were found

Reply:
We could change this check to just be:

```java
if (accountsList == null) {
    // ...
}
```

Since the `foreach` loop will basically skip over an empty `List`